# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/vr/VRManager.h
# Commit: ec6a64b639a8
# Full Hash: ec6a64b639a82e957376c09a71aef7ce62b3a5aa
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-12-03 09:47:26
# Regressor Bug: 1657404
# File Overlap Count: 3
# Description:
#   Bug 1657404 - Implement a strongly-typed, service-dependent gamepad handle r=handyman,aklotz
#   
#   Currently, the gamepad code uses a uint32_t as a handle and does some trickery
#   with it by trying to create a unique ID and adding an offset to it for VR code.
#   
# ==============================================================================

diff -r 6792f28b99bd -r ec6a64b639a8 gfx/vr/VRManager.h
--- a/gfx/vr/VRManager.h	Wed Dec 02 22:36:35 2020 +0000
+++ b/gfx/vr/VRManager.h	Wed Dec 02 23:06:05 2020 +0000
@@ -12,6 +12,7 @@
 #include "nsTArray.h"
 #include "nsThreadUtils.h"
 #include "mozilla/Atomics.h"
+#include "mozilla/dom/GamepadHandle.h"
 #include "mozilla/layers/LayersSurfaces.h"  // for SurfaceDescriptor
 #include "mozilla/Monitor.h"
 #include "mozilla/TimeStamp.h"
@@ -54,10 +55,10 @@
   void EnumerateDevices();
   void StopAllHaptics();
 
-  void VibrateHaptic(uint32_t aControllerIdx, uint32_t aHapticIndex,
-                     double aIntensity, double aDuration,
+  void VibrateHaptic(mozilla::dom::GamepadHandle aGamepadHandle,
+                     uint32_t aHapticIndex, double aIntensity, double aDuration,
                      const VRManagerPromise& aPromise);
-  void StopVibrateHaptic(uint32_t aControllerIdx);
+  void StopVibrateHaptic(mozilla::dom::GamepadHandle aGamepadHandle);
   void NotifyVibrateHapticCompleted(const VRManagerPromise& aPromise);
   void StartVRNavigation(const uint32_t& aDisplayID);
   void StopVRNavigation(const uint32_t& aDisplayID,